### ISSUE TITLE
Individual years for +2C climate futures & Abatzoglou, J.T., S.Z. Dobrowski, S.A. Parks, K.C. Hegewisch

### ISSUE THREAD

**Issue by nileshsnshinde** [samapriya/awesome-gee-community-datasets#34]
### Dataset Name
Individual years for +2C and +4C climate futures
TerraClimate layers commensurate with global mean temperatures +2C and +4C above preindustrial levels. These data are available for pseudo years 1985-2015. Future climate projections were developed for two different climate futures: (1) when global mean temperatures are 2C warmer than pre-industrial, and (2) when global mean temperatures are 4C above preindustrial. This data is at two links;
+2C data: http://thredds.northwestknowledge.net:8080/thredds/catalog/TERRACLIMATE_ALL/data_plus2C/catalog.html 
+4C data: 
http://thredds.northwestknowledge.net:8080/thredds/catalog/TERRACLIMATE_ALL/data_plus4C/catalog.html 
About TerraClimate
TerraClimate: Monthly Climate and Climatic Water Balance for Global Terrestrial Surfaces, University of Idaho is already exiting in GEE here (https://developers.google.com/earth-engine/datasets/catalog/IDAHO_EPSCOR_TERRACLIMATE ). It is a dataset of monthly climate and climatic water balance for global terrestrial surfaces from 1958-2019. These data provide important inputs for ecological and hydrological studies at global scales that require high spatial resolution and time-varying data. All data have monthly temporal resolution and a ~4-km (1/24th degree) spatial resolution. The data cover the period from 1958-2020. We plan to update these data periodically (annually).
Extra Info: Feel free to add any optional details 
Paper: Abatzoglou, J.T., S.Z. Dobrowski, S.A. Parks, K.C. Hegewisch, 2018, Terraclimate, a high-resolution global dataset of monthly climate and climatic water balance from 1958-2015, Scientific Data
Website: https://www.climatologylab.org/terraclimate.html 
Data link: http://thredds.northwestknowledge.net:8080/thredds/catalog/TERRACLIMATE_ALL/data_plus2C/catalog.html 
http://thredds.northwestknowledge.net:8080/thredds/catalog/TERRACLIMATE_ALL/data_plus4C/catalog.html 
Download Tool/Code snippets if any: NA

#### Required
License Information: To the extent possible under law, John Abatzoglou has waived all copyright and related or neighboring rights to TerraClimate. This work is published from: United States

Curated by: Abatzoglou, J.T., S.Z. Dobrowski, S.A. Parks, K.C. Hegewisch,

Keywords: Individual years for climate futures under +2C and +4C from TerraClimate 

Last updated: April 15, 2021

**Comment by nileshsnshinde** [comment]
Hi Dr. Roy, 
Under the TerraClimate data, they have provided this information on page "guide for dataset abbreviations" , here https://www.climatologylab.org/terraclimate-variables.html. 

I am listing the variable information herewith, 
The following variables are provided for download as 30-year climatological monthly summaries or monthly data for each year (1958-present). Data are available on [NKN THREDDS servers](http://thredds.northwestknowledge.net:8080/thredds/catalog/TERRACLIMATE_ALL/catalog.html) or through Google Earth Engine. Note that data come as compressed netCDF. Be sure to reference scale_factor and offset commands when looking at the data.

aet (Actual Evapotranspiration, monthly total), units = mm
def (Climate Water Deficit, monthly total), units = mm
pet (Potential evapotranspiration, monthly total), units = mm
ppt (Precipitation, monthly total), units = mm
q (Runoff, monthly total), units = mm
soil (Soil Moisture, total column - at end of month), units = mm
srad (Downward surface shortwave radiation), units = W/m2
swe (Snow water equivalent - at end of month), units = mm
tmax (Max Temperature, average for month), units = C
tmin (Min Temperature, average for month), units = C
vap (Vapor pressure, average for month), units  = kPa
ws (Wind speed, average for month), units = m/s
vpd (Vapor Pressure Deficit, average for month), units = kpa
PDSI (Palmer Drought Severity Index, at end of month), units = unitless

Please let me know if you have any additional questions.

I appreciate your work and inputs here. 

Best, 
Nilesh

**Comment by samapriya** [comment]
Hi @nileshsnshinde 
Just to get a bit of a sanity check could you try this collection to make sure it looks okay I need to attach the offset and scale as metadata to each collections as well but wanted to do a quick check
https://code.earthengine.google.com/cab4b572b1cb6bd0ebf68841b6e73fd6

**Comment by nileshsnshinde** [comment]
Hi Dr. Roy, 
I tried using the collection, it is working! However, the values are not exactly what we expected, maybe this is because of the scale or projection. Please advice. 

Best, 
Nilesh

**Comment by samapriya** [comment]
Hi @nileshsnshinde 
I used a simple gdal_translate with lzw compression. If you have a conversion code for .nc to .tif that you currently use let me know I am happy to rerun the datasets

**Comment by nileshsnshinde** [comment]
Hi Dr. Roy, 
I think this is "simple gdal_translate with lzw compression" an apt approach, I am using the same.  I am not sure why the max temperature under 2C scenario is varying from .nc (from website  http://thredds.northwestknowledge.net:8080/thredds/catalog/TERRACLIMATE_ALL/data_plus2C/catalog.html ) and from an asset you shared with me earlier. 

Please see the following table of 1985 Tmax observations for Delhi, 
 from .nc file 1985 in Celsius | from  .nc file 1985 in Kelvin | from asset shared by you 1985 
-- | -- | --
27.2 | 300.35 | 822
34.5 | 307.65 | 875
38.5 | 311.65 | 950
42.4 | 315.55 | 988
41 | 314.15 | 1026
35.3 | 308.45 | 1015
35.8 | 308.95 | 956
35.4 | 308.55 | 954
32.7 | 305.85 | 957
29.9 | 303.05 | 903
24.8 | 297.95 | 850



Thank you, 
Nilesh

**Comment by samapriya** [comment]
Hi @nileshsnshinde 
Can you provide me the lat long for the place you created the table for ? This will help me do the checks

Also try tmax for 4C here is the collection 
projects/sat-io/open-datasets/TERRACLIMATE/4C/tmax

**Comment by nileshsnshinde** [comment]
Hi Dr. Roy, 
I wrote the following script for quick overview of both dataset. I think 4c version looks fine but the 2c version max temperature is different. 

*script

//roi
var point = /* color: #d63000 */ee.Geometry.Point([77.19352795668044, 28.629300008353113]);

//data on TerraClimate 2c and 4c for month October
var dataset2c = ee.ImageCollection("projects/sat-io/open-datasets/TERRACLIMATE/2C/tmax").select('b10');
var dataset4c = ee.ImageCollection("projects/sat-io/open-datasets/TERRACLIMATE/4C/tmax").select('b10');
var maximumTemperatureVis = {"opacity":1,"bands":["b10"],"min":409,"max":1002,
"palette":["1a3678","2955bc","5699ff","8dbae9","acd1ff","caebff","e5f9ff","fdffb4","ffe6a2","ffc969","ffa12d","ff7c1f","ca531a","ff0000","ab0000"]};

//print the contents
print(dataset2c,{},'tmax2c');
print(dataset4c,{},'tmax4c');

//add layer 
Map.addLayer(dataset2c, maximumTemperatureVis, 'Maximum Temperature 2C');
Map.addLayer(dataset4c, maximumTemperatureVis, 'Maximum Temperature 4C');
Map.addLayer(point,{},'location:delhi');


// Define the chart and print it to the console.
var chart =
    ui.Chart.image
        .seriesByRegion({
          imageCollection: dataset2c,
          band: 'b10',
          regions: point,
          reducer: ee.Reducer.mean(),
          scale: 4000,
          seriesProperty: 'label',
          xProperty: 'system:time_start'
        })
        .setOptions({
          title: 'dataset2c-1985-october(b10)',
          hAxis: {title: 'Date', titleTextStyle: {italic: false, bold: true}},
          vAxis: {
            title: 'tmax (?)',
            titleTextStyle: {italic: false, bold: true}
          },
          lineWidth: 5,
          colors: ['f0af07', '0f8755', '76b349'],
        });
print(chart);

// Define the chart and print it to the console.
var chart =
    ui.Chart.image
        .seriesByRegion({
          imageCollection: dataset4c,
          band: 'b10',
          regions: point,
          reducer: ee.Reducer.mean(),
          scale: 4000,
          seriesProperty: 'label',
          xProperty: 'system:time_start'
        })
        .setOptions({
          title: 'dataset4c-1985-october(b10)',
          hAxis: {title: 'Date', titleTextStyle: {italic: false, bold: true}},
          vAxis: {
            title: 'tmax (?)',
            titleTextStyle: {italic: false, bold: true}
          },
          lineWidth: 5,
          colors: ['f0af07', '0f8755', '76b349'],
        });
print(chart);

**Comment by samapriya** [comment]
Thanks @nileshsnshinde 
I downloaded the tmax again to check for this incase the source file had any issues I am not sure what is going on here. Can you download a sample .nc file to test again
http://thredds.northwestknowledge.net:8080/thredds/fileServer/TERRACLIMATE_ALL/data_plus2C/TerraClimate_2c_tmax_1985.nc

I almost wonder if they uploaded incorrect files to the server. At this point I have tried downloading , reprojecting to EPSG 4326 and so on. 

I also just provided you access to all 2C variables and 4C variables for quick test. For example
just replace with variable name like swe, soil and so on 

projects/sat-io/open-datasets/TERRACLIMATE/2C/ppt

I look forward to your findings. I was also wondering if it would be possible for you to upload the NC files you have for tmax/tmin for 2C to a google drive and share a link ? I would like to convert the version you have and test it out

**Comment by nileshsnshinde** [comment]
Hi Dr. Roy, 
I have uploaded the on Google drive. I share it with you in your email (samapriya.roy@gmail.com). 
Sorry for the delay, my internet is too slow. 

Best, 
Nilesh

**Comment by samapriya** [comment]
Hi @nileshsnshinde 
Okay I guess all it needed was a weekend of actually looking at the data, I think I figured it out we needed to add the scale and offset to get to the base Celsius values check out this script and let me know if it makes sense. Again note the temperature in the graph is now in Celsius

https://code.earthengine.google.com/4dad3dd9b7db73fbda086590ff17f920

I think this makes way more sense now and is aligned to expected values. If that is the case I will reprocess and add scale and offset as metadata. Also I am bounding the results so the collections appear better in GEE with bounds and the collection is a bounded collection
